pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/RealtimeSegmentAssignment.java

```diff
@@ -125,8 +133,13 @@ private List<String> assignConsumingSegment(String segmentName, InstancePartitio
       return instancesAssigned;
     } else {
       // Replica-group based assignment
-
-      checkReplication(instancePartitions);
+      // TODO: Refactor check replication this for segment assignment strategy in follow up PR
```

Comment:
nit: link the OSS issue here. Same for other places with `TODO`.

Reply:
updated